admin-password-updater/updater/event_handler.go

```diff
@@ -0,0 +1,172 @@
+				oldPasswd := adminSection.Key("password").String()
+
+				if defaultUser != adminUser {
+					u.Log.V(0).Info("exiting because usernames do not match",
+						"default-user", defaultUser, "default-user-file", u.DefaultUserFile,
+						"admin-user", adminUser, "admin-file", u.AdminFile)
+					u.Done <- true
+					return
+				}
+				if newPasswd == oldPasswd {
+					u.Log.V(2).Info("passwords already match, nothing to do", "username", defaultUser)
+					break
+				}
+
+				u.Rmqc.SetUsername(adminUser)
+				u.Rmqc.SetPassword(oldPasswd)
+
+				if !u.updateInRabbitMQ(adminUser, newPasswd) {
+					break
+				}
+
+				u.Log.V(4).Info("copying new password...", "source", u.DefaultUserFile, "target", u.AdminFile)
+				adminSection.Key("password").SetValue(newPasswd)
+				if err := adminCfg.SaveTo(u.AdminFile); err != nil {
+					u.Log.Error(err, "failed to write new password", "file", u.AdminFile)
+					u.Done <- true
+					return
+				}
+				u.Log.V(2).Info("copied new password", "source", u.DefaultUserFile, "target", u.AdminFile)
+			} else {
+				u.Log.V(4).Info("file system event", "file", event.Name, "operation", event.Op.String())
+			}
+
+		case err, ok := <-u.Watcher.Errors:
+			if !ok {
+				u.Log.V(0).Info("watcher errors channel is closed, exiting...")
+				u.Done <- true
+				return
+			}
+			u.Log.Error(err, "failed to watch", "directory", u.WatchDir)
+		}
+	}
+
+}
+
+func fileChanged(defaultUserFile string, event fsnotify.Event) bool {
+	return event.Name == defaultUserFile &&
+		(event.Op&fsnotify.Create == fsnotify.Create ||
+			event.Op&fsnotify.Write == fsnotify.Write)
+}
+
+// updateInRabbitMQ sets newPasswd for existingUser in the RabbitMQ server.
+// It returns true if password could be sucessfully updated or password was already up-to-date.
+func (u *PasswordUpdater) updateInRabbitMQ(existingUser, newPasswd string) (ok bool) {
+	pathUsers := "/api/users/" + existingUser
+
+	user, err := u.Rmqc.GetUser(existingUser)
+	if err != nil {
+		return u.handleHTTPError(err, http.MethodGet, pathUsers, newPasswd)
+	}
+
+	// We succeeded to fetch user tags, continue to update user password.
+	newUserSettings := rabbithole.UserSettings{
+		Name:             existingUser,
+		Tags:             user.Tags,
+		Password:         newPasswd,
+		HashingAlgorithm: user.HashingAlgorithm,
+	}
+	resp, err := u.Rmqc.PutUser(existingUser, newUserSettings)
+	if err != nil {
+		return u.handleHTTPError(err, http.MethodPut, pathUsers, newPasswd)
+	}
+
+	u.Log.V(3).Info("HTTP response", "method", http.MethodPut, "path", pathUsers, "status", resp.Status)
+	u.Log.V(2).Info("updated password on RabbitMQ server", "user", existingUser)
+	return true
+}
+
+func (u *PasswordUpdater) handleHTTPError(err error, httpMethod, pathUsers, newPasswd string) (ok bool) {
+	// as returned in
+	// https://github.com/michaelklishin/rabbit-hole/blob/1de83b96b8ba1e29afd003143a9d8a8234d4e913/client.go#L153
+	if err.Error() == "Error: API responded with a 401 Unauthorized" {
```

Comment:
Very good call. I thought about the same. We could use https://github.com/michaelklishin/rabbit-hole/blob/a692ae424d424bff3b5c9a8bbd97f4f362756a0f/error.go#L5-L14. However, I fear we might break the rabbithole API by doing so since other users might depend on that specific error message (as for example implemented in here)? WDYT, should we risk it? I'm happy either way.